unittests/osr_test.py

```diff
@@ -0,0 +1,153 @@
+# Copyright (c) 2017 CEF Python, see the Authors file.
+# All rights reserved. Licensed under BSD 3-clause license.
+# Project website: https://github.com/cztomczak/cefpython
+"""OSR testing of CEF Python."""
+
+import unittest
+import _test_runner
+from cefpython3 import cefpython as cef
+import platform
+import sys,time,os
+import base64
+
+g_datauri_data = """
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title>Title</title>
+    <script>
+        function print(msg) {
+            console.log(msg + " [JS]");
+        }
+        window.onload = function(){
+            document.addEventListener("mousedown", function( e ) {
+              print('mousedown:'+e.pageX+","+e.pageY)
+            }, false);
+            document.addEventListener("mouseup", function( e ) {
+              print('mouseup:'+e.pageX+","+e.pageY)
+            }, false);
+
+            document.addEventListener("mousemove", function( e ) {
+              print('mousemove:'+e.pageX+","+e.pageY)
+            }, false);
+
+            document.addEventListener("dragenter", function( e ) {
+              print('dragenter:'+e.pageX+","+e.pageY)
+            }, false);
+
+            document.addEventListener("dragover", function( e ) {
+              print('dragover:'+e.pageX+","+e.pageY)
+            }, false);
+            document.addEventListener("drop", function( e ) {
+              print('drop:'+e.pageX+","+e.pageY)
+            }, false);
+
+
+        }
+        print('window.onload init')
+    </script>
+    <style>
+        .box{
+            background-color: #F00;
+            height: 200px;
+        }
+    </style>
+</head>
+<body>
+
+<input id="selectFile" type="file">
+<div class="box"></div>
+</body>
+</html>
+
+"""
+
+g_datauri = "data:text/html;base64," + base64.b64encode(g_datauri_data.encode(
+    "utf-8", "replace")).decode("utf-8", "replace")
+VIEWPORT_SIZE = (1024, 768)
+
+g_subtests_ran = 0
+
+def subtest_message(message):
+    global g_subtests_ran
+    g_subtests_ran += 1
+    print(str(g_subtests_ran) + ". " + message)
+    sys.stdout.flush()
+
+class Handler(object):
+    def __init__(self):
+        self.OnPaint_called = False
+
+    def OnConsoleMessage(self, message, **_):
+        if "error" in message.lower() or "uncaught" in message.lower():
+            self.javascript_errors_False = True
+            raise Exception(message)
+        else:
+            # Check whether messages from javascript are coming
+            self.OnConsoleMessage_True = True
+            subtest_message(message)
+
+    def OnDragEnter(self, browser, dragData, mask):
```

Comment:
We should keep the naming convention for handlers as in main_test.py, so that there are DragHandler (with OnDragEnter and StartDragging) and LoadHandler (with OnLoadEnd) classes.